### PR TITLE
Workshop reminder one week before and three days before

### PR DESCRIPTION
**Pull request by rigoucr**

### CODE REVIEW

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc [hunk 1]
```diff
@@ -96,3 +96,77 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
   $wrapper->save();
 }
 
+/**
+ * Implements hook_cron().
+ */
+function dccr_ftr_workshop_cron() {
+  $week_before = strtotime('-7 day',DCCR_CAMP_START_DATE);
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc]: space after comma; please

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc [hunk 2]
```diff
@@ -96,3 +96,77 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
...
+  $three_days_before = strtotime('-3 day',DCCR_CAMP_START_DATE);
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc]: same here

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc [hunk 3]
```diff
@@ -96,3 +96,77 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
...
+ if (DCCR_CAMP_START_DATE >= REQUEST_TIME && $week_before <= REQUEST_TIME && !variable_get('REMINDER_WEEK_BEFORE', FALSE)) {
+    variable_get('REMINDER_WEEK_BEFORE', TRUE);
+    dccr_ftr_workshop_invoke_reminde('7');
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc]: I think this is a typo (you missed the letter 'r')

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc [hunk 4]
```diff
@@ -96,3 +96,77 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
   $wrapper->save();
 }
 
+/**
+ * Implements hook_cron().
+ */
+function dccr_ftr_workshop_cron() {
+  $week_before = strtotime('-7 day',DCCR_CAMP_START_DATE);
+  $three_days_before = strtotime('-3 day',DCCR_CAMP_START_DATE);
+ if (DCCR_CAMP_START_DATE >= REQUEST_TIME && $week_before <= REQUEST_TIME && !variable_get('REMINDER_WEEK_BEFORE', FALSE)) {
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc]: First condition is not necessary.
variable name should be prefixed with module name and follow variable names conventions (no uppercase)

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc [hunk 5]
```diff
@@ -96,3 +96,77 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
...
+    variable_get('REMINDER_WEEK_BEFORE', TRUE);
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc]: I think this should be a variable_set

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc [hunk 6]
```diff
@@ -96,3 +96,77 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
...
+function dccr_ftr_workshop_invoke_reminder($days) {
+  $result = db_select('node', 'n')
+    ->fields('n',array('nid'))
+    ->condition('type','workshop', '=')
+    ->condition('status',1, '=')
+    ->execute();
+  $workshops_nids = $result->fetchCol();
+  $workshops = node_load_multiple($workshops_nids);
+
+  foreach ($workshops_nids as $nid) {
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc:138]: node_load_multiple returns an array keyed by nid; so I think you could iterate over $workshops `foreach ($workshops as $nid => $workshop)`

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc [hunk 7]
```diff
@@ -96,3 +96,77 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
...
+    $users_uids = dccr_ftr_workshop_get_workshop_flaggers($nid);
+    $users = user_load_multiple($users_uids);
+    foreach ($users_uids as $uid) {
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc:141]: same note about node_load_multiple applies for user_load_multiple

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc [hunk 8]
```diff
@@ -96,3 +96,77 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
...
+      rules_invoke_event('dccr_ftr_workshop_nofications', $workshops[$nid],$users[$uid], $days);
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc]: comma after parameter please.

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc [hunk 9]
```diff
@@ -96,3 +96,77 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
...
+  return $result->fetchCol();
+}
+
+/**
+ * Implements hook_rules_event_info().
+ */
+function dccr_ftr_workshop_rules_event_info() {
+  return array(
+    'dccr_ftr_workshop_nofications' => array(
+      'label' => t('Time to execute Workshop email reminder'),
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc:165]: Is this label accurate? (only as a question)

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.strongarm.inc [hunk 10]
```diff
@@ -13,6 +13,13 @@ function dccr_ftr_workshop_strongarm() {
   $strongarm = new stdClass();
   $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
   $strongarm->api_version = 1;
+  $strongarm->name = 'date_format_workshop_reminder';
+  $strongarm->value = 'l, F dS, Y \\a\\t G:i';
+  $export['date_format_workshop_reminder'] = $strongarm;
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.strongarm.inc:18]: I need to test this; but I think date formats doesn't get well exported through features

**Comment by kporras07** [comment]
I added some inline notes; some of them are about coding standards; but there is a couple of them about functionality. Please fix them and I'll do a functional test.

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc (hunk 1 above)
Comment (rigoucr) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc]: Ready

**Comment by rigoucr** [comment]
@kporras07 thanks for your comments. 
I fixed the errors and the coding standards, now you can test the functionality.
:+1:

File: sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc [hunk 11]
```diff
@@ -96,3 +96,106 @@ function dccr_ftr_workshop_countdown($action, $content_id) {
   $wrapper->save();
 }
 
+/**
+ * Implements hook_cron().
+ */
+function dccr_ftr_workshop_cron() {
+  $week_before = strtotime('-7 day', DCCR_CAMP_START_DATE);
+  $three_days_before = strtotime('-3 day', DCCR_CAMP_START_DATE);
+ if ($week_before <= REQUEST_TIME && !variable_get('reminder_week_before', FALSE)) {
```
Comment (kporras07) [inline sites/all/modules/features/dccr_ftr_workshop/dccr_ftr_workshop.module.inc]: Please fix indentation of this block.

**Comment by kporras07** [comment]
Only one new comment.
When you fix that; please merge.

**Pull request merged**